### PR TITLE
Move selectors into their own directory

### PR DESCRIPTION
**Pull request by amyanchen**
Previously, selectors lived in the container files they were used in. Now, they all live in the selectors directory, along with their corresponding tests.

### CODE REVIEW

**Comment by amyanchen** [comment]
[Move selectors into their own directory](https://trello.com/c/IvFg0oN6/72-move-selectors-into-their-own-directory)

File: src/selectors/index.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { meetsDuringDow, meetsDuringHour } from 'components/calendar/calendar-helpers';
```
Comment (Joonpark13) [inline src/selectors/index.js]: Now that these are being used outside of components, let's put these in `util/time` instead and import them from there.

File: src/selectors/index.js [hunk 2]
```diff
@@ -17,3 +19,113 @@ export function scheduledSectionsSelector(state) {
 export function selectedCourseNameSelector(state) {
   return state.getIn(['browse', 'selected', 'course', 'name']);
 }
+
```
Comment (Joonpark13) [inline src/selectors/index.js:18]: Can we move all of the helper functions to the top, then order the selectors below that by top level store fields? Like all the `search` selectors, then all the `browse` selectors, etc.

Comment (Joonpark13) [reply src/selectors/index.js:18]: Wouldn't hurt to delineate each of the three using comments either (see action-types.js)

**Review by Joonpark13 — CHANGES_REQUESTED** [review]

File: src/selectors/index.js [hunk 3]
```diff
@@ -1,7 +1,57 @@
...
+  return classes.find(
+    (sectionOrAssociatedClass) => {
+      const eventObj = sectionOrAssociatedClass.get('event').toJS();
+      return meetsDuringDow(eventObj, dow) && meetsDuringHour(eventObj, hour);
+    },
+  );
+}
+
+/* schedule */
+export function scheduledSectionsSelector(state) {
```
Comment (Joonpark13) [inline src/selectors/index.js]: This should be `schedule` not `scheduled`

File: src/selectors/index.js [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
+function findMatchingClass(classes, hour, dow) {
+  return classes.find(
+    (sectionOrAssociatedClass) => {
+      const eventObj = sectionOrAssociatedClass.get('event').toJS();
+      return meetsDuringDow(eventObj, dow) && meetsDuringHour(eventObj, hour);
+    },
+  );
+}
+
+/* schedule */
```
Comment (Joonpark13) [inline src/selectors/index.js]: No need for this to be a multi line comment. Can we make this `// Schedule` instead? (Same with browse and search)

**Review by Joonpark13 — CHANGES_REQUESTED** [review]

**Comment by Joonpark13** [comment]
Also seeing a console error on load - can you look into that? Make sure you manually load up and test the app for your PRs!

**Comment by Joonpark13** [comment]
The console error on load is gone but now it errors out when you add a class.

**Review by Joonpark13 — APPROVED** [review]

**Review by Joonpark13 — CHANGES_REQUESTED** [review]
Oh I just noticed you have commits under `images` - not sure how you got that in there but don't think we need those.

**Review by Joonpark13 — APPROVED** [review]

**Pull request merged**